src/payment/strategies/googlepay/googlepay-payment-processor.ts

```diff
@@ -232,4 +233,12 @@ export default class GooglePayPaymentProcessor {
             this._billingAddressActionCreator.updateAddress(googlePayAddressMapped)
         );
     }
+
+    private _getFirstAndLastName(name: string[]): string[] {
+        if (name.length > 1) {
+            return [name.slice(0, -1).join(' '), ...name.slice(-1)];
+        }
+
+        return [name.join(' '), ''];
```

Comment:
In this scenario we're rejoining a string we've only just split, by starting with the whole name still as a string we can simply reuse it:

```
    private _getFirstAndLastName(fullName: string): [string, string] {
        const nameParts = fullName.split(' ');

        if (nameParts.length === 1) {
            return [fullName, ''];
        }

        const firstName = nameParts.slice(0, -1).join(' ');
        const lastName = nameParts[nameParts.length - 1];

        return [firstName, lastName];
    }
```

Reply:
This function also deserves its own tests, given that is covers a few scenarios

```
_getFirstAndLastName('Lee Dunkley');
_getFirstAndLastName('Cher');
_getFirstAndLastName('Neil deGrasse Tyson');
```

Reply:
Why not also use it in `_mapGooglePayAddressToShippingAddress` ?